src/txadmission.cpp

```diff
@@ -253,31 +243,76 @@ void CommitTxToMempool()
             SyncWithWallets(data.entry.GetSharedTx(), nullptr, -1);
 #endif
             vWhatChanged.push_back(data.hash);
-
             // Update txn per second only when a txn is valid and accepted to the mempool
             mempool.UpdateTransactionsPerSecond();
         }
         txCommitQ.clear();
     }
 
+    std::map<uint256, CTxInputData> mapWasDeferred;
     {
         LOCK(csTxInQ);
         // Clear the filter of incoming conflicts, and put all queued tx on the deferred queue since they've been
         // deferred
+        LOG(MEMPOOL, "txadmission incoming filter reset.  Current txInQ size: %d\n", txInQ.size());
         incomingConflicts.reset();
         while (!txInQ.empty())
         {
             txDeferQ.push(txInQ.front());
             txInQ.pop();
         }
-        LOG(MEMPOOL, "Reset incoming filter\n");
+
+        // Move the previously deferred txns into active processing.
+
+        // We MUST push the first item in the defer queue in without checking it against incoming conflicts.
+        // This is fine because the first insert into an empty incomingConflicts must succeed.
```

Comment:
nit:  maybe something like

-We MUST push the first item in the defer queue in without checking 
+We MUST push the first item in the defer queue to input queue without checking